keras_nlp/layers/position_embedding.py

```diff
@@ -0,0 +1,119 @@
+      seq_axis: The axis of the input tensor where we add the embeddings.
+
+    Example:
+    ```python
+    class PositionalEmbedding(layers.Layer):
+        def __init__(self, max_length, vocab_size, embed_dim, **kwargs):
+            super().__init__(**kwargs)
+            self.token_embeddings = layers.Embedding(
+                input_dim=vocab_size, output_dim=embed_dim
+            )
+            self.position_embeddings = keras_nlp.layers.PositionEmbedding(
+                max_length=max_length
+            )
+
+        def call(self, inputs):
+            embedded_tokens = self.token_embeddings(inputs)
+            embedded_positions = self.position_embeddings(inputs)
+            return embedded_tokens + embedded_positions
+    ```
+
+    Reference:
+      [BERT: Pre-training of Deep Bidirectional Transformers for Language
+      Understanding](https://arxiv.org/abs/1810.04805).
+    """
+
+    def __init__(
+        self,
+        max_length,
+        initializer="glorot_uniform",
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        if max_length is None:
+            raise ValueError("`max_length` must be an Integer, not `None`.")
+        self.max_length = int(max_length)
+        self.initializer = keras.initializers.get(initializer)
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "max_length": self.max_length,
+                "initializer": keras.initializers.serialize(self.initializer),
+            }
+        )
+        return config
+
+    def build(self, input_shape):
+        feature_size = input_shape[-1]
+        self.position_embeddings = self.add_weight(
+            "embeddings",
+            shape=[self.max_length, feature_size],
+            initializer=self.initializer,
+            trainable=True,
+        )
+
+        super().build(input_shape)
+
+    def call(self, inputs):
+        # the dimension for the sequence is the one before the last
+        sequence_rank = -2
+
+        if isinstance(inputs, tf.RaggedTensor):
+            # first trim position_embeddings to the max sequence length
+            bounding_shape = inputs.bounding_shape()
+            max_sequence_length = bounding_shape[sequence_rank]
+            position_embeddings = self.position_embeddings[
+                :max_sequence_length, :
+            ]
+            # then broadcast to add the missing dimensions
+            position_embeddings = tf.broadcast_to(
+                position_embeddings, bounding_shape
+            )
+            # then apply row lengths to recreate the same ragged shape as inputs
+            return tf.RaggedTensor.from_tensor(
+                position_embeddings,
+                inputs.row_lengths(),
+                ragged_rank=len(bounding_shape) + sequence_rank,
+            )
+        else:
+            # first trim position_embeddings to the actual sequence length
+            input_shape = tf.shape(inputs)
```

Comment:
You could pull from this line up to broadcast_to into a helper function used here and ragged. Maybe just one local to the call function.

def reshape_embedding(target_shape):

Reply:
Will add utility method

Reply:
Oh, and I'll and a check verifying that the ragged dimension is the expected one